pynicotine/gtkgui/userbrowse.py

```diff
@@ -687,6 +694,13 @@ def on_select_dir(self, selection):
 
         self.set_directory(iterator.user_data)
 
+    def on_select_file(self, selection):
+
+        if selection is None:
+            return
+
+        self.select_files()
```

Comment:
I think I would be more comfortable having something like `self.num_selected_files = selection.count_selected_rows()` here instead of requesting data for every row on each selection.

Reply:
You're right that was inefficient as this is not needed for every case where this is called. Thank you for your suggestion.

Reply:
This conversation comment has become misplaced. This was in reference to on_select_file which has been implemented with `selection.count_selected_rows()` on line 775 as per your suggestion.